airbyte-integrations/bases/debezium/src/main/java/io/airbyte/integrations/debezium/CdcTargetPosition.java

```diff
@@ -13,23 +14,17 @@
  * we might end up syncing forever. In order to tackle that, we need to define a point to end at the
  * beginning of the sync
  */
-public interface CdcTargetPosition {
+public interface CdcTargetPosition<T> {
 
   /**
    * Reads a position value (lsn) from a change event and compares it to target lsn
    *
    * @param valueAsJson json representation of a change event
    * @return true if event lsn is equal or greater than targer lsn, or if last snapshot event
    */
-  boolean reachedTargetPosition(JsonNode valueAsJson);
+  boolean reachedTargetPosition(final JsonNode valueAsJson);
 
-  /**
-   * Checks if a specified lsn has reached the target lsn.
-   *
-   * @param lsn an lsn value
-   * @return true if lsn is equal or greater than target lsn
-   */
-  default boolean reachedTargetPosition(final Long lsn) {
+  default boolean reachedTargetPosition(final T positionFromHeartbeat) {
```

Comment:
Should we just remove the default?
It just throw an exception as it's an interface and all the classes needs to implement it.
Otherwise we can make default the method `extractPositionFromHeartbeatOffset`.